RubberduckTests/Inspections/UnreachableCaseInspectionTests.cs

```diff
@@ -0,0 +1,2398 @@
+        [TestCase("IsLT=45,IsGT=205,Range=20:70,Single=200", "IsLT=45,IsGT=205,Range=20:70", "Single=200")]
+        [TestCase("Range=60:80", "Range=20:70,Range=65:100", "")]
+        [TestCase("Range=60:80", "IsLT=100", "")]
+        [TestCase("Range=60:80", "IsGT=1", "")]
+        [TestCase("Single=17", "Range=1:4,Range=7:9,Range=15:20", "")]
+        [TestCase("Single=17", "IsLT=45", "")]
+        [TestCase("Single=17", "IsGT=-45000", "")]
+        [TestCase("Single=17,Single=20", "Single=16,Single=17,Single=18,Single=19", "Single=20")]
+        [TestCase("Range=101:149", "Range=101:149,Range=1:100", "")]
+        [TestCase("RelOp=x < 50", "Single=-1,Single=0", "")]
+        [TestCase("RelOp=x < 50", "Single=-1, RelOp=x < 50", "")]
+        [Category("Inspections")]
+        public void UciUnit_RemovalRangeClauses(string candidateClauseInput, string existingClauseInput, string expectedClauses)
+        {
+            var sumClauses = RangeDescriptorsToFilters(new string[] { candidateClauseInput, existingClauseInput }, Tokens.Long);
+            var clausesToFilter = sumClauses[0];
+            var filter = sumClauses[1];
+
+            var filterResults = RangeClauseFilterFactory.Create(Tokens.Long, ValueFactory);
+
+            filterResults = clausesToFilter.FilterUnreachableClauses(filter);
+            if (filterResults.ContainsFilters)
+            {
+                var clauses = expectedClauses.Split(new string[] { "," }, StringSplitOptions.None);
+                var expected = CreateTestFilter(clauses.ToList(), Tokens.Long);
+                Assert.AreEqual(expected, filterResults);
+            }
+            else
+            {
+                if (!expectedClauses.Equals(""))
+                {
+                    Assert.Fail("Function fails to return Interface");
+                }
+            }
+        }
+
+        [TestCase("Range=0:10", "Single=50", "Boolean")]
+        [TestCase("Range=True:False", "Single=50", "Boolean")]
+        [TestCase("Range=False:True", "Single=50", "Boolean")]
+        [TestCase("Single=-5000", "Single=False", "Boolean")]
+        [TestCase("Single=True", "Single=0", "Boolean")]
+        [TestCase("Single=500", "Single=0", "Boolean")]
+        [TestCase("IsLT=5", "IsGT=-5000", "Long")]
+        [TestCase("IsLT=40,IsGT=40", "Range=35:45", "Long")]
+        [TestCase("IsLT=40,IsGT=44", "Range=35:45", "Long")]
+        [TestCase("IsLT=40,IsGT=40", "Single=40", "Long")]
+        [TestCase("IsGT=240,Range=150:239", "Single=240, Single=0,Single=1,Range=2:150", "Byte")]
+        [TestCase("Range=151:255", "Single=150, Single=0,Single=1,Range=2:149", "Byte")]
+        [TestCase("IsLT=13,IsGT=30,Range=30:100", "Single=13,Single=14,Single=15,Single=16,Single=17,Single=18,Range=19:30", "Long")]
+        [Category("Inspections")]
+        public void UciUnit_FiltersAll(string firstCase, string secondCase, string typeName)
```

Comment:
I'm having hard time comprehending this unit test. Why would the `firstcase` be `Range=0:10`, `secondCase`, `Single=50`, with `Boolean`?

Reply:
Again, this is a theoretical case where the user has a Select Case Expression that is of the Boolean type, but they have Case Statements that are expressed as `Longs`.  In the end, each case is evaluated as Boolean.  So, Range 0:10 covers both True and False - and those are the only two values that actually matter - every other number is just noise.